demos/inpage/src/helpers/encodeUtils.ts

```diff
@@ -0,0 +1,115 @@
+export function hexJoin(hexStrings: string[]) {
+  return `0x${hexStrings.map(remove0x).join('')}`;
+}
+
+export function hexLen(hexString: string) {
+  return (hexString.length - 2) / 2;
+}
+
+export function remove0x(hexString: string) {
+  if (!hexString.startsWith('0x')) {
+    throw new Error('Expected 0x prefix');
+  }
+
+  return hexString.slice(2);
+}
+
+export function encodeVLQ(xParam: bigint) {
+  let x = xParam;
+  const segments: bigint[] = [];
+
+  while (true) {
+    const segment = x % 128n;
+    segments.unshift(segment);
+    x -= segment;
+    x /= 128n;
+
+    if (x === 0n) {
+      break;
+    }
+  }
+
+  let result = '0x';
+
+  for (let i = 0; i < segments.length; i++) {
+    const keepGoing = i !== segments.length - 1;
```

Comment:
Nah I copied them from `bls-wallet`.

Not sure what you mean by whether `keepGoing` is conceptually `lastIndex`. If you're referring to [`lastIndexOf`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/lastIndexOf), then no. If you just mean whether `keepGoing` is a boolean that says whether `i` is the last index, then yes.

Conceptually, `keepGoing` makes a bit more sense with the decoder in mind - each byte uses the high bit as a flag for the decoder to indicate whether it should keep going. This allows us to avoid using a dedicated byte (or more) to encode the length, allowing for 0-127 to be encoded in a single byte.